internal/storage/db/internal/db.go

```diff
@@ -478,7 +480,72 @@ func (s *dbStorage) GetDependents(ctx context.Context, ids ...namer.ModuleID) (m
 	return out, nil
 }
 
-func (s *dbStorage) Disable(ctx context.Context, policyKey ...string) (int64, error) {
+func (s *dbStorage) HasDescendants(ctx context.Context, ids ...namer.ModuleID) (map[namer.ModuleID]bool, error) {
+	// SELECT 1
+	// FROM policy_ancestor pa1 JOIN policy p1 ON (pa1.policy_id = p1.id AND p1.disabled = false)
+	// WHERE pa1.ancestor_id = p.id;
+	innerQuery := s.db.Select(goqu.L("1")).
+		From(goqu.T(PolicyAncestorTbl).As("pa1")).
+		Join(
+			goqu.T(PolicyTbl).As("p1"),
+			goqu.On(
+				goqu.And(
+					goqu.C(PolicyAncestorTblPolicyIDCol).Table("pa1").Eq(goqu.C(PolicyTblIDCol).Table("p1")),
+					goqu.C(PolicyTblDisabledCol).Table("p1").Eq(goqu.V(false)),
+				),
+			),
+		).
+		Where(goqu.C(PolicyAncestorTblAncestorIDCol).Table("pa1").Eq(goqu.C(PolicyTblIDCol).Table("p")))
+
+	// SELECT p.id, EXISTS(<innerQuery>) AS has_descendants
+	// FROM policy p
+	// WHERE p.id IN (?);
+	query := s.db.Select(
+		goqu.C(PolicyTblIDCol).Table("p"),
+		goqu.L("EXISTS ?", innerQuery).As("has_descendants"),
```

Comment:
Can we use [`Func`](https://pkg.go.dev/github.com/doug-martin/goqu/v9#Func) here instead? I think that would convey the idea better.


```suggestion
		goqu.Func("EXISTS", innerQuery).As("has_descendants"),
```

Reply:
`goqu.Func` puts an extra pair of parentheses and postgres doesn't seem to like it.